libindy/src/api/anoncreds.rs

```diff
@@ -55,6 +57,21 @@ pub extern fn indy_issuer_create_schema(command_handle: i32,
 
     trace!("indy_issuer_create_schema: entity >>> issuer_did: {:?}, name: {:?}, version: {:?}, attrs: {:?}", issuer_did, name, version, attrs);
 
+    let attrs: HashSet<String> = match serde_json::from_str(&attrs)
```

Comment:
@AxelNennker I suggest to stop this PR on conversion of jsons to domain types. This early validation is a big question and will make this PR hard to merge.

Reply:
ok

Reply:
Any estimate one when there will be a decision on the big question?

As I said in the description the PR I think that this greatly improves understanding for developers using the API.

Reply:
In my vision we can call STATIC-only validation in API layer, but this validation must be defined in domain layer. I suggest the following:

- We can can define trait Validatable with one method validate() -> Result<(), Error> in utils/validation
- For each domain struct that requires static validation we can implement this trait
- We can define additional macro check_usefull_valid_json! that will also call validate method on domain struct

@jovfer @AxelNennker What do you think?

One addition that i suggest is to return position error instead of CommonInvalidStructure here, but we need consensus here as it is breaking change.

Reply:
Also we need consider to remove static validation from command and service layers in this case

Reply:
I would do this step by step.

Before this PR the API checked for valid cstr which this PR keeps and should keep.
What this PR adds is adding `check_useful_json!` which uses `serde_json::from_str::<$t>(&$x)` to parse the json string $x to the domain type $t.

I think this validation is enough for this step.

Not sure about a new trait. I haven'd tried it but serde_json might be all we need already and then implement from_str for the domain type... which can do additional validation steps on the instance variable.

// What is position error? The api only returns error code.